alcove/index/language.py

```
@@ -0,0 +1,299 @@
+
+def get_language_detector(provider: str | None = None) -> LanguageDetector:
+    from alcove.config import load_config
+    from alcove.plugins import discover_language_detectors
+
+    cfg = load_config().language
+    choice = _normalize_language(provider or cfg.provider)
+    detectors = dict(_BUILTIN_LANGUAGE_DETECTORS)
+    detectors.update(discover_language_detectors())
+    cls = detectors.get(choice)
+    if cls is None:
+        raise ValueError(f"Unknown language detector: {choice!r}.")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)